python/k4FWCore/ApplicationMgr.py

```diff
@@ -75,34 +136,12 @@ def fix_properties(self):
             or (props["Output"][0] and props["Output"][0] != "<no value>")
         ):
             writer = Writer("k4FWCore__Writer")
+
         # Let's tell the Reader one of the input files so it can
         # know which collections it's going to read
         if reader is not None:
-            # Open the files and get the number of events. This is necessary to
-            # avoid errors when running multithreaded since if we have, for
-            # example, 10 events and we are running 9 at the same time, then
-            # (possibly) the first 9 complete and 9 more are scheduled, out of
-            # which only one will be finished without errors. If we know the
-            # number of events in advance then we can just schedule those.
-            inp = None
-            if props["input"][0]:
-                inp = "input"
-            elif props["Input"][0]:
-                inp = "Input"
-            if inp:
-                # Import here to avoid always importing ROOT which is slow
-                from podio.root_io import Reader as PodioReader
-
-                podio_reader = PodioReader(props[inp][0])
-                if self._mgr.EvtMax == -1:
-                    self._mgr.EvtMax = podio_reader._reader.getEntries("events")
-                try:
-                    frame = podio_reader.get("events")[0]
-                except IndexError:
-                    print("Warning, the events category wasn't found in the input file")
-                    raise
-                collections = list(frame.getAvailableCollections())
-                reader.InputCollections = collections
```

Comment:
This was always running when there was input, otherwise the Reader doesn't have information about which collections is going to read which is needed as a property because of all the stuff (creating the handles that will be used to push data) that happens before `initialize()`.